public/app/plugins/datasource/prometheus/result_transformer.ts

```diff
@@ -41,12 +42,25 @@ interface TimeAndValue {
   [TIME_SERIES_VALUE_FIELD_NAME]: number;
 }
 
+const isTableResult = (dataFrame: DataFrame, options: DataQueryRequest<PromQuery>): boolean => {
+  // We want to process instant results in Explore as table
+  if ((options.app === CoreApp.Explore && dataFrame.meta?.custom?.queryType) === 'instant') {
```

Comment:
I am just wondering could we figure out if data frame contains instant vector data just from the data itself? Like if there is only single timestamp? Or is that too ambiguous? I assume even range query can return a single value?

Reply:
Hmmm, one more thing I came up with is passing queryResults format instead of querytype in data Frame. That could be better in a sense,  that we are not passing information from query, but from result.

Reply:
What do you think?